src/agent/communicator/CMakeLists.txt

```diff
@@ -1,13 +1,43 @@
-cmake_minimum_required(VERSION 3.12.4)
-
-project(Communicator)
+cmake_minimum_required(VERSION 3.24)
 
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_BUILD_TYPE Release)
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_BUILD_TYPE RelWithDebInfo)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
+
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_SOURCE_DIR}/../../vcpkg/scripts/buildsystems/vcpkg.cmake")
+set(VCPKG_MANIFEST_DIR ${CMAKE_SOURCE_DIR}/../../)
 
-# Enable testing
-enable_testing()
+project(Communicator)
+message(STATUS "******* Project Communicator *********")
+
+message(STATUS "CMAKE_SOURCE_DIR: ${CMAKE_SOURCE_DIR}")
+message(STATUS "CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
 
 include_directories(include)
-include_directories(${CMAKE_SOURCE_DIR}/communicator)
+
+find_package(Boost REQUIRED COMPONENTS system url)
+if(Boost_FOUND)
+    include_directories(${Boost_INCLUDE_DIRS})
+    link_directories(${Boost_LIBRARY_DIRS})
+endif()
+
+find_package(nlohmann_json CONFIG REQUIRED)
+find_package(OpenSSL REQUIRED)
```

Comment:
Here we add openssl for the first time, but the vcpkg manifest hasn't been changed. It needs to be added in `vcpkg.json` too.

Reply:
Done.